src/Runtime/OMExternalConstant.inc

```diff
@@ -74,6 +74,7 @@ void omMMapBinaryFile(
   memcpy(tPath, fname, tLen);
   __a2e_s(tPath);
   fname = tPath;
+  fname[tLen] = '\0';
```

Comment:
Do we need to increase `tLen` by 1 so that the allocated buffer can hold the extra `\0`?

Reply:
Looking at the generated code, `filename` already has `\0`. I guess the issue is that `strlen(fname)` does not count `\0` and when we did `memcpy`, `\0` was missed.

Could you try to increase `tLen` by one? e.g.
```
  char *tPath;
  size_t tLen = strlen(fname) + 1;
  tPath = (char *)malloc(tLen);
```

Reply:
Using `strdup` can simplify the code. We have similar code in `src/Runtime/jni/jniwrapper.c`:
```
#ifdef __MVS__
/* On z/OS, we convert entry point name in ASCII into EBCDIC for
 * the omInputSignature/omOutputSignaturee function using __a2e_s.
 * However, __a2e_s converts in place so we need to make a copy of
 * the entry point name.
 *
 * On z/OS, we convert signature in EBCDIC into ASCII for the JNI
 * NewStringUTF function using __e2a_s. However, __e2a_s converts
 * in place so we need to make a copy of the signature.
 */
#define __a2e(s) conv(s, __a2e_s)
#define __e2a(s) conv(s, __e2a_s)
static inline char *conv(const char *str, size_t (*fptr)(char *)) {
  char *dup = strdup(str);
  if (dup == NULL)
    return NULL;

  /* Convert ASCII to EBCDIC in place */
  if (fptr(dup) < 0) {
    free(dup);
    return NULL;
  }

  return dup;
}
#endif
```

Reply:
I followed this suggestion, thank you.